code/rdb/rdbstandard.q

```diff
@@ -1,5 +1,27 @@
 // Get the relevant RDB attributes
-.proc.getattributes:{`date`tables!(.rdb.rdbpartition[],();tables[])}
+.proc.getattributes:{default:`date`tables`procname!(.rdb.rdbpartition[];tables[] except .rdb.ignorelist;.proc.procname);
+    / get all cols that contains date (of type "pdz")
+    timecolumns:1!{tcols:exec c from meta value x where t in"pdz";
+        (enlist[`tablename]!enlist x),
+            /functional select to get the min value (defaults to `timestamp$.z.d for starttimestamp)
+            enlist[`timecolumns]!enlist?[x;();();
+                tcols!(enlist,/:enlist each($;enlist`timestamp),/:enlist each((?),/:enlist each(=;0W),/:mtcols),'`.z.d,'mtcols:enlist each min,/:tcols),\:0Wp]}each tables[`.] except .rdb.ignorelist;
+    / update date attribute for .gw.partdict and .gw.attributesrouting
+    default[`date]:asc default[`date]union first[d]+til 1+last deltas d:exec(min;max)@\:distinct`date$raze[value each timecolumns][;0]from timecolumns;
+        / for striped rdbs, retrieve stripe mapping from stp and deduce instruments in each table to report as attributes to the gateway 
+    stphandle:$[count i:.servers.getservers[`proctype;`segmentedtickerplant;()!();1b;0b];
+        first exec w from i;
+            .lg.e[`.proc.getattributes[];"Failed to get a valid handle to the segmented tickerplant process"]];
+    .ds.tblstripe:@[stphandle;"select tbl,filts from .stpps.subrequestfiltered where handle = .z.w";
+        {.lg.e[`.proc.getattributes;"Failed to retrieve stripe map from STP"]}];
```

Comment:
do we need to do this every time we update attributes? Can we not store this when we subscribe & just reference it here? (I don't think this info will change at any time other than subscription)